pkg/controller/common/reconciler.go

```diff
@@ -87,9 +88,111 @@ func (r *BaseIngressReconciler) ReconcileIngress(ctx context.Context, ci network
 	return nil
 }
 
+func (r *BaseIngressReconciler) reconcileNetworkPolicy(ctx context.Context, ci networkingv1alpha1.IngressAccessor) error {
+	desired := resources.MakeNetworkPolicyAllowAll(ci)
+
+	var networkPolicyList networkingv1.NetworkPolicyList
+	if err := r.Client.List(ctx, &client.ListOptions{}, &networkPolicyList); err != nil {
+		return err
+	}
+
+	// Detect if the user has any NetworkPolicy objects in this namespace
+	foundUserNetworkPolicy := false
+	for _, networkPolicy := range networkPolicyList.Items {
+		// Don't treat the NetworkPolicy we create as user-created
+		if networkPolicy.Name == desired.Name && networkPolicy.Namespace == desired.Namespace {
+			continue
+		}
+
+		// Don't treat the NetworkPolicy owned by Service Mesh as user-created
+		if networkPolicy.Labels["maistra.io/owner"] != "" {
+			continue
+		}
+
+		// Treat any other NetworkPolicy we find as user-created
+		foundUserNetworkPolicy = true
```

Comment:
Should we break out of the for-loop here?

Reply:
In fact, you can just `return nil` here and move the comment from below in here.